bridge/src/worker.rs

```diff
@@ -46,80 +38,54 @@ pub enum Response {
 pub type WorkerResult = Result<Vec<u8>, WorkerError>;
 type WorkerCallback = Box<dyn FnOnce(WorkerResult) + Send + 'static>;
 
-async fn handle_request(request: Request, outer_channel: SyncSender<Response>) {
-    match request {
-        Request::PollActivityTask { core_worker, callback } => {
-            let task = core_worker.poll_activity_task().await;
+async fn poll_activity_task_async(worker: Worker, callback: WorkerCallback) {
+    let task = worker.core_worker.poll_activity_task().await;
 
-            let response = match task {
-                Ok(task) => {
-                    let mut bytes: Vec<u8> = Vec::with_capacity(task.encoded_len());
-                    task.encode(&mut bytes).expect("Unable to encode activity task protobuf");
+    let response = match task {
+        Ok(task) => {
+            let mut bytes: Vec<u8> = Vec::with_capacity(task.encoded_len());
+            task.encode(&mut bytes).expect("Unable to encode activity task protobuf");
 
-                    Response::ActivityTask { bytes, callback }
-                },
-                Err(e) => {
-                    Response::Error { error: WorkerError::UnableToPollActivityTask(e), callback }
-                }
-            };
-
-            outer_channel.send(response).expect("Unable to send a callback");
+            Response::ActivityTask { bytes, callback }
```

Comment:
You would wrap the callback here and move the bytes into the new closure to do what I'm talking about.

This callback type would remain the same, but the one in the enum would change to be parameterless.